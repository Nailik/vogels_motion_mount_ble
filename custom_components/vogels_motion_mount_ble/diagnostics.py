"""Diagnostics support for Vogels Motion Mount BLE."""

from __future__ import annotations

from typing import Any

from homeassistant.components.diagnostics import async_redact_data
from homeassistant.core import HomeAssistant

from .const import CONF_PIN, CONF_MAC
from .coordinator import VogelsMotionMountBleCoordinator
from . import VogelsMotionMountBleConfigEntry

TO_REDACT = {CONF_PIN, CONF_MAC}


async def async_get_config_entry_diagnostics(
    _: HomeAssistant, config_entry: VogelsMotionMountBleConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    coordinator: VogelsMotionMountBleCoordinator = config_entry.runtime_data

    return {
        "config_entry_data": async_redact_data(dict(config_entry.data), TO_REDACT),
        "vogels_motion_mount_ble_data": coordinator.data,
    }
