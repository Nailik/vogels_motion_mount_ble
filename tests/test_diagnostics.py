"""Tests for the diagnostics data provided by the Vogels Motion Mount (BLE) integration."""

from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant

from pytest_homeassistant_custom_component.common import MockConfigEntry
from pytest_homeassistant_custom_component.components.diagnostics import (
    get_diagnostics_for_config_entry,
)
from pytest_homeassistant_custom_component.typing import ClientSessionGenerator


async def test_diagnostics(
    hass: HomeAssistant,
    hass_client: ClientSessionGenerator,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test diagnostics for config entry."""
    mock_config_entry.add_to_hass(hass)
    assert await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()
    diag = await get_diagnostics_for_config_entry(hass, hass_client, mock_config_entry)
    assert diag == snapshot
