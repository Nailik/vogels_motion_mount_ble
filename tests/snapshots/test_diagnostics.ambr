# serializer version: 1
# name: test_diagnostics
  dict({
    'config_entry_data': dict({
      'conf_mac': '**REDACTED**',
      'conf_name': 'Mount',
      'conf_pin': '**REDACTED**',
    }),
    'vogels_motion_mount_ble_data': dict({
      '__type': "<class 'custom_components.vogels_motion_mount_ble.data.VogelsMotionMountData'>",
      'repr': "VogelsMotionMountData(automove=<VogelsMotionMountAutoMoveType.Hdmi_1_On: 0>, available=True, connected=True, distance=100, freeze_preset_index=0, multi_pin_features=VogelsMotionMountMultiPinFeatures(change_default_position=True, change_name=True, change_presets=True, change_tv_on_off_detection=False, disable_channel=False, start_calibration=True), name='Living Room Mount', pin_setting=<VogelsMotionMountPinSettings.Single: 13>, presets=[VogelsMotionMountPreset(index=0, data=VogelsMotionMountPresetData(distance=80, name='Movie Mode', rotation=15)), VogelsMotionMountPreset(index=1, data=VogelsMotionMountPresetData(distance=80, name='Movie Mode', rotation=15)), VogelsMotionMountPreset(index=2, data=VogelsMotionMountPresetData(distance=80, name='Movie Mode', rotation=15)), VogelsMotionMountPreset(index=3, data=VogelsMotionMountPresetData(distance=100, name='Gaming Mode', rotation=-10)), VogelsMotionMountPreset(index=4, data=VogelsMotionMountPresetData(distance=80, name='Movie Mode', rotation=15)), VogelsMotionMountPreset(index=5, data=VogelsMotionMountPresetData(distance=80, name='Movie Mode', rotation=15)), VogelsMotionMountPreset(index=6, data=VogelsMotionMountPresetData(distance=80, name='Movie Mode', rotation=15))], rotation=5, tv_width=140, versions=VogelsMotionMountVersions(ceb_bl_version='1.0.0', mcp_bl_version='1.0.1', mcp_fw_version='2.0.0', mcp_hw_version='revA'), permissions=VogelsMotionMountPermissions(auth_status=VogelsMotionMountAuthenticationStatus(auth_type=<VogelsMotionMountAuthenticationType.Full: 2>, cooldown=None), change_settings=True, change_default_position=True, change_name=True, change_presets=True, change_tv_on_off_detection=True, disable_channel=False, start_calibration=True), requested_distance=None, requested_rotation=None)",
    }),
  })
# ---
